src/autoloads/resources.gd

```diff
@@ -0,0 +1,71 @@
+extends Node
+
+## HOW TO USE THE RESOURCE MANAGER
+## Example: get a list of resources in a set of directories
+## Dictionary of directories, with the namespace, followed by the directory path:
+#	var dirs: Dictionary = {
+#	"Resource 1": "res://examples/resource1",
+#	"Resource 2": "res://examples/resource2"
+#	}
+#
+## Array of file types to use (only files with these extensions will be listed):
+#	var file_types: Array = [".png", ".svg"]
+#
+## Pass the two variables into the list function:
+#	Resources.list(dirs, file_types)
+#	> {Resource 1:{Test 1:test_1.png, Test 2:test_2.svg}, Resource 2:{Test 3:test_3.svg}}
+#
+## Select only resources in "Resource 1":
+#	Resources.list(dirs, file_types)["Resource 1"]
+#	> {Test 1:test_1.png, Test 2:test_2.svg}
+
+# --Public Functions--
+
+# Returns a dictionary of all resources from a given directory dictionary
+func list(directories: Dictionary, types: Array) -> Dictionary:
```

Comment:
could set the type of `types` to PoolStringArray to denote that it should be a list of strings